src/test/java/org/kohsuke/github/GHWorkflowRunTest.java

```diff
@@ -0,0 +1,223 @@
+                    assertEquals(workflow.getId(), workflowRun.getWorkflowId());
+                    assertTrue(workflowRun.getUrl().getPath().contains("/actions/runs/"));
+                    assertTrue(workflowRun.getHtmlUrl().getPath().contains("/actions/runs/"));
+                    assertTrue(workflowRun.getJobsUrl().getPath().endsWith("/jobs"));
+                    assertTrue(workflowRun.getLogsUrl().getPath().endsWith("/logs"));
+                    assertTrue(workflowRun.getCheckSuiteUrl().getPath().contains("/check-suites/"));
+                    assertTrue(workflowRun.getArtifactsUrl().getPath().endsWith("/artifacts"));
+                    assertTrue(workflowRun.getCancelUrl().getPath().endsWith("/cancel"));
+                    assertTrue(workflowRun.getRerunUrl().getPath().endsWith("/rerun"));
+                    assertTrue(workflowRun.getWorkflowUrl().getPath().contains("/actions/workflows/"));
+                    assertEquals(MAIN_BRANCH, workflowRun.getHeadBranch());
+                    assertNotNull(workflowRun.getHeadCommit().getId());
+                    assertNotNull(workflowRun.getHeadCommit().getTreeId());
+                    assertNotNull(workflowRun.getHeadCommit().getMessage());
+                    assertNotNull(workflowRun.getHeadCommit().getTimestamp());
+                    assertNotNull(workflowRun.getHeadCommit().getAuthor().getEmail());
+                    assertNotNull(workflowRun.getHeadCommit().getCommitter().getEmail());
+                    assertEquals(GHEvent.WORKFLOW_DISPATCH, workflowRun.getEvent());
+                    assertEquals(Status.COMPLETED, workflowRun.getStatus());
+                    assertEquals(Conclusion.SUCCESS, workflowRun.getConclusion());
+                    assertNotNull(workflowRun.getHeadSha());
+
+                    return true;
+                }
+            }
+            return false;
+        });
+    }
+
+    @Test
+    public void testCancelAndRerun() throws IOException {
+        GHWorkflow workflow = repo.getWorkflow(SLOW_WORKFLOW_PATH);
+
+        long latestPreexistingWorkflowRunId = getLatestPreexistingWorkflowRunId();
+
+        workflow.dispatch(MAIN_BRANCH);
+
+        // now that we have triggered the workflow run, we will wait until it's in progress and then cancel it
+        Awaitility.await().atLeast(atLeast).pollInterval(pollInterval).atMost(atMost).until(() -> {
+            List<GHWorkflowRun> workflowRuns = getLatestWorkflowRuns(MAIN_BRANCH, Status.IN_PROGRESS);
+            for (GHWorkflowRun workflowRun : workflowRuns) {
+                if (workflowRun.getName().equals(SLOW_WORKFLOW_NAME)
+                        && workflowRun.getId() > latestPreexistingWorkflowRunId) {
+                    assertNotNull(workflowRun.getId());
+
+                    workflowRun.cancel();
+                    cancelledWorkflowRunId = workflowRun.getId();
+                    return true;
+                }
+            }
+            return false;
+        });
```

Comment:
A few requests applying to all the `await` sections: 
1. I'd prefer if the `until` only polled for the state change and the asserts are called after, outside the `until`.  
2.  Let's use `getGitHubBeforeAfter()` for polling.  That way the tests don't need to record the polling.  That way when running CI, the test do effectively no waiting.  
3.  With (2) we can remove the special polling interval values in `setup()`.  Maybe use a [fibonacci polling interval](https://github.com/awaitility/awaitility/wiki/Usage#fibonacci-poll-interval) with milliseconds? Recording will burn a few extra requests but they won't be recorded.

Reply:
1. yeah that was my inclination as well first and I thought it would be better to reduce the number of API calls. I changed it. I agree it's more readable this way.
2. ah, yes, good idea, I was looking for a way to do exactly that
3. I removed the polling when we don't use a proxy. I tried the fibonacci thingy but it doesn't work very well, don't know why but it was just running forever and I don't really have the cycles to dig more into it. Using a fixed poll interval works so I am sticking with that.

Reply:
Thanks, looks great. Bummer about the polling interval, but oh well.